src/predictor/gpu_predictor.cu

```diff
@@ -324,21 +336,27 @@ class GPUPredictor : public xgboost::Predictor {
                 h_nodes.begin() + h_tree_segments[tree_idx - tree_begin]);
     }
 
-    size_t i_batch = 0;
     size_t batch_offset = 0;
     for (auto &batch : dmat->GetRowBatches()) {
-      CHECK(i_batch == 0 || devices_.Size() == 1) << "External memory not supported for multi-GPU";
-      // out_preds have been sharded and resized in InitOutPredictions()
-      batch.offset.Shard(GPUDistribution::Overlap(devices_, 1));
+      GPUSet devices = GPUSet::All(param_.gpu_id, param_.n_gpus, batch.Size());
+      ConfigureShards(devices);
+
+      batch.offset.Reshard(GPUDistribution::Overlap(devices_, 1));
+
       std::vector<size_t> device_offsets;
       DeviceOffsets(batch.offset, &device_offsets);
       batch.data.Reshard(GPUDistribution::Explicit(devices_, device_offsets));
+
+      std::vector<size_t> out_preds_offsets;
```

Comment:
by changing the distribution of `out_preds` this way, callers expectation of the distribution within this host device vector - its granularity, size etc. may be violated *after* the predict api is invoked for any post processing such as transformation via objective function etc.

Reply:
If we need the whole vector back on GPUs for further processing, we can always reshard it.

Reply:
wouldn't this break existing code? let's say if the user does this
```
predictor->predict(out_predictions)
get_gradients(out_predictions, gradients)
do_tree_boost(gradients);
```
if the follow-up steps implicitly assumed the predictions to be shared based on number_of_rows/classes etc., will this change not break that code without the user having to recode to reshard?

Reply:
Add a final resharding step to preserve the current behavior.